File: Ruby/selectionsort.rb
def selection_sort(arr)
    n = arr.length
  
    (0...n - 1).each do |i|
      min_index = i
  
      # Find the index of the minimum element in the remaining unsorted part
      (i + 1...n).each do |j|
        min_index = j if arr[j] < arr[min_index]
      end
  
      # Swap the found minimum element with the first element
      arr[i], arr[min_index] = arr[min_index], arr[i]
    end
  end
  
  # Example usage
  arr = [64, 25, 12, 22, 11]
  puts "Unsorted array: #{arr}"
  
  selection_sort(arr)
  
  puts "Sorted array: #{arr}"
  

File: Go/selectionsort.go
package main

import "fmt"

func selectionSort(arr []int) {
    n := len(arr)

    for i := 0; i < n-1; i++ {
        // Find the minimum element in the unsorted part of the array
        minIndex := i
        for j := i + 1; j < n; j++ {
            if arr[j] < arr[minIndex] {
                minIndex = j
            }
        }

        // Swap the found minimum element with the first element
        arr[i], arr[minIndex] = arr[minIndex], arr[i]
    }
}

func main() {
    // Example usage
    arr := []int{64, 25, 12, 22, 11}
    fmt.Println("Unsorted array:", arr)

    selectionSort(arr)

    fmt.Println("Sorted array:", arr)
}
